src/system/NetworkNamespace.cpp

```diff
@@ -43,6 +41,8 @@ std::shared_ptr<NetworkNamespace> claimNetworkNamespace()
                 namespaces.emplace_back(
                   std::make_shared<NetworkNamespace>(netnsName));
             }
+
+            namespacesInitialised = true;
```

Comment:
Interesting, I've not look at this code for a long time, but could we just have a single `UniqueLock` that covers the whole function? There shouldn't be too much contention on this function, so should be ok for performance.